Integrations/src/main/java/io/deephaven/integrations/python/PythonReplayListenerAdapter.java

```diff
@@ -64,11 +41,21 @@ public PythonReplayListenerAdapter(String description, Table source, PyObject py
      * @param source The source table to which this listener will subscribe.
      * @param retain Whether a hard reference to this listener should be maintained to prevent it from being collected.
      * @param pyObjectIn Python listener object.
+     * @param dependencies The tables that must be satisfied before this listener is executed.
      */
-    public PythonReplayListenerAdapter(String description, Table source, boolean retain,
-            PyObject pyObjectIn) {
+    private PythonReplayListenerAdapter(String description, Table source, boolean retain, PyObject pyObjectIn,
+            Table... dependencies) {
         super(description, source, retain);
-        pyCallable = PythonUtils.pyListenerFunc(pyObjectIn);
+        this.dependencies = dependencies;
+        this.pyCallable = PythonUtils.pyListenerFunc(pyObjectIn);
+    }
+
+    public static PythonReplayListenerAdapter create(String description, Table source, boolean retain,
```

Comment:
I would put this static method *above* the constructor. This is the one reader's should see first.